partials/AmqpPublisher.java

```diff
@@ -0,0 +1,38 @@
+{% macro amqpPublisher(asyncapi, params) %}
+
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+{% for channelName, channel in asyncapi.channels() %}
+    {%- if channel.hasSubscribe() %}
+        {%- for message in channel.subscribe().messages() %}
+import {{params['userJavaPackage']}}.model.{{message.payload().uid() | camelCase | upperFirst}};
+        {%- endfor -%}
+    {% endif -%}
+{% endfor %}
+
+
+@Service
+public class PublisherService {
+    @Autowired
+    private RabbitTemplate template;
+
+    {% for channelName, channel in asyncapi.channels() %}
```

Comment:
Taking into account issue on line 29-30 I would expect that here we generate variables only for channels with subscribe api